preciceconfigchecker/rule.py

```diff
@@ -7,31 +9,28 @@ class Severity(Enum):
 
 
 class Rule:
-    severity:Severity
-    name:str
-    message:str
-    possible_solutions:str
-    followed:bool = True
-    numbers_not_followed:int = 0
-
-    def __init__(self, severity:Severity = Severity.INFO, name:str = "Rule", message:str = "", possible_solutions:str = "") -> None:
+    def __init__(self, problem:Problem, severity:Severity = Severity.INFO) -> None:
+        self.problem = problem
         self.severity = severity
-        self.name = name
-        self.message = message
-        self.possible_solutions = possible_solutions
+        self.violations:List[Violation] = []
+        self.followed = False
 
-    def check_with(self, check_method, args) -> None:
-        result = check_method(*args)
-        if type(result) is not bool:
-            print("Error : The passed method does not return a bool value!")
+    def check_with(self, check_method:Callable, args:tuple) -> None:
```

Comment:
In my opinion, having it calculated in the constructor is not a good idea. You should first initialize it and then calculate it.
It would work in itself, but it's not good style.

Reply:
I'll take a look at making the abstract class.

Reply:
When we create an abstract class, we need to write a separate class for each rule type so that we can override that one method. I think that making the class abstract is not a good idea.

The idea was that we usually pass one main method, but also another method if necessary. This is easier in my opinion.

Another idea would be that the class gets static methods. Or you can define the methods in 'rule.py', but outside the 'Rule' class, it should be the same. Then we define an enum 'rule type' that uses the appropriate method. However, arguments still have to be passed for each rule, which are then passed on to the method for the appropriate rule type.

Info: 'func()' is something like a static method, 'func(self)' is not.

Reply:
If we are 100% sure that we only use this one method, that would be our 'check' method.

Reply:
Do we want to discuss in our group what makes more sense?